### PR TITLE
Added [script]: rename-file by modified time

### LINKED ISSUE #2312: [Script]: Rename files by files modified time
### Aim

In this script, we can rename files by modified time. For example, if there is 50+ files in a directory and we have to sort them according to their creation time or last modified time, we can directly use this script.

### Details

In this script, we use a special type of comparator inside the sort function to make this work. After that, we use os module to make the rename happen. The script is written following the data structures and algorithms and is optimized as much as possible.

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSoC'23 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by Rancho2002**
# Description
This software can rename your current directory files in ascending order based on modification. Oldest file named with 01_oldname .... like this

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update
- [x] Documentation Update

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] I have added the project meta data in the PR template.
- [x] I have created the ``requirements.txt`` file if needed.

## Project Metadata

None

### CODE REVIEW

File: Rename_by_time/main.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        else:
+            oldName=files[i]
+
+        rename+=1
+        newfile_name=f"{str(rename).zfill(digits)}_{oldName}"
+
+        if(n=="p"):
+            print(f"{oldName} --> {newfile_name}")
+        else:
+            os.rename(oldName,newfile_name)
```
Comment (JRudransh) [inline Rename_by_time/main.py:37]: Before rename, check if new file name is already exist or not to avoid exception

**Review by JRudransh — CHANGES_REQUESTED** [review]

File: Rename_by_time/main.py (hunk 1 above)
Comment (Rancho2002) [reply Rename_by_time/main.py:37]: Hello @JRudransh , the code is written in that way if the new file name is already present for example 01_filename before the code run, it automatically change the filename before renaming. For references, pls check this screenshot I have attached. 

### Before adding the new file name I am checking what can be the new file name could be
Screenshot from 2023-07-22 09-58-30


### As you can check 01_vid1.c is one of the file name , so I make it already present in the directory and run the code. As you can check in preview, how it manage the existing same new file name by making it 13_01_filename

image

**Comment by Rancho2002** [comment]
Please check the code now, and tell if it required any additional changes @JRudransh @avinashkranjan

**Review by JRudransh — APPROVED** [review]
Okay, When you do listdir in `main.py:15` You are reversing that list of dirs, that's why `01_vid1.c` renamed to `13_01_vid1.c`  first and than `vid1.c`  remaned to `01_vid1.c` and you didn't get any error. I think the code won't break for now.

**Comment by Rancho2002** [comment]
hello @JRudransh , can you pls merge it ?

**Comment by JRudransh** [comment]
Sorry I dont have the privilage to merge it, @avinashkranjan merge it

**Review by avinashkranjan — APPROVED** [review]

**Pull request merged**